include/envoy/stats/stats.h

```diff
@@ -263,10 +303,20 @@ class Store : public Scope {
    * @return a list of all known gauges.
    */
   virtual std::list<GaugeSharedPtr> gauges() const PURE;
+
+  /**
+   * @return a list of all known histograms.
+   */
+  virtual std::list<HistogramSharedPtr> histograms() const PURE;
 };
 
 typedef std::unique_ptr<Store> StorePtr;
 
+/**
+ * Callback invoked when a store's mergeHistogram() runs.
+ */
+typedef std::function<void()> PostMergeCb;
```

Comment:
nit: I think we can probably kill the typedef for this, just use std::function inline below.

Reply:
I have added it because, it is repeated in tests as well. if you are strongly think about removing it, I can remove it.